dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java

```diff
@@ -6,57 +6,160 @@
+import com.dotmarketing.business.*;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
 import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.portlets.contentlet.model.ContentletDependencies;
 import com.dotmarketing.portlets.folders.business.FolderAPI;
 import com.dotmarketing.portlets.workflows.business.BaseWorkflowIntegrationTest;
 import com.dotmarketing.portlets.workflows.business.SystemWorkflowConstants;
+import com.dotmarketing.portlets.workflows.business.WorkFlowFactory;
 import com.dotmarketing.portlets.workflows.business.WorkflowAPI;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
 import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
 import com.dotmarketing.tag.model.TagInode;
+import com.github.rjeschke.txtmark.Run;
+import com.liferay.portal.model.User;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.jetbrains.annotations.NotNull;
 import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 
+import static com.dotcms.util.CollectionsUtils.set;
+
+@RunWith(DataProviderRunner.class)
 public class SaveContentActionletTest extends BaseWorkflowIntegrationTest {
 
-    private static WorkflowAPI workflowAPI = null;
-    private static ContentTypeAPI contentTypeAPI = null;
-    private static ContentType customContentType = null;
-    private static final int LIMIT = 20;
+
+    private static User systemUser = APILocator.systemUser();
+    private static List<ContentType> contentTypes = new ArrayList<>();
+
+    private static class TestCase {
+        private final boolean respectFrontendRoles;
+        private final User user;
+        private final boolean hasSaveActionPermission;
+        private final boolean hasContentTypeAddChildrenPermission;
+        private final ContentType contentType;
+
+        private TestCase(
+                final boolean respectFrontendRoles,
+                final User user,
+                final ContentType contentType,
+                final boolean hasSaveActionPermission,
+                final boolean hasContentTypeAddChildrenPermission) {
+            this.respectFrontendRoles = respectFrontendRoles;
+            this.user = user;
+            this.hasSaveActionPermission = hasSaveActionPermission;
+            this.hasContentTypeAddChildrenPermission = hasContentTypeAddChildrenPermission;
+            this.contentType = contentType;
+        }
+    }
+
+    @DataProvider
+    public static Object[] usersAndContentTypeWithoutHostField() throws Exception {
+        final WorkflowAPI workflowAPI = APILocator.getWorkflowAPI();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)